subsys/net/lib/tftp/tftp_client.c

```diff
@@ -155,8 +155,11 @@ static inline int send_err(int sock, struct tftpc *client, int err_code, char *e
 
 	/* Copy the Error String. */
 	if (err_msg != NULL) {
-		strcpy(client->tftp_buf + req_size, err_msg);
-		req_size += strlen(err_msg);
+		size_t copy_len = MIN(sizeof(client->tftp_buf) - req_size,
+				      strlen(err_msg));
```

Comment:
The `MIN` macro might have the effect of calling it twice.

```suggestion
		size_t copy_len = strlen(err_msg);
		if (copy_len > sizeof(client->tftp_buf) - req_size) {
			copy_len = sizeof(client->tftp_buf) - req_size;
		}
```

Reply:
Thanks, updated as suggested.